### PR TITLE
Search events implemented

### PR DESCRIPTION
**Pull request by MasterRadule**

### CODE REVIEW

File: Backend/src/main/java/ktsnvt/tim1/repositories/EventRepository.java [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
 import ktsnvt.tim1.model.Event;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 
 public interface EventRepository extends JpaRepository<Event, Long> {
+    Event findOneByName(String name);
+
+    @Query(value = "select * from event e where lower(e.name) like ?1 and e.category like ?2 and cast(location_id as char) like ?3", nativeQuery = true)
```
Comment (draganculibrk9) [inline Backend/src/main/java/ktsnvt/tim1/repositories/EventRepository.java]: Query can be converted to method with keywords

File: Backend/src/main/java/ktsnvt/tim1/controllers/EventController.java [hunk 2]
```diff
@@ -38,8 +39,32 @@
...
+        catch (EntityNotFoundException e) {
+            return new ResponseEntity<>(e.getMessage(), HttpStatus.NOT_FOUND);
+        }
+        catch (Exception e) {
+            return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @GetMapping(value = "search")
+    public ResponseEntity<Object> searchEvents(@Valid @RequestBody SearchEventsDTO searchDTO, Pageable pageable) {
```
Comment (draganculibrk9) [inline Backend/src/main/java/ktsnvt/tim1/controllers/EventController.java]: @RequestBody shouldn't be used in @GetMapping

File: Backend/src/main/java/ktsnvt/tim1/repositories/EventRepository.java (hunk 1 above)
Comment (MasterRadule) [reply Backend/src/main/java/ktsnvt/tim1/repositories/EventRepository.java]: The case is that if location is null then all events which meet other criteria should be returned. I can not do that with keywords and one method.

Comment (draganculibrk9) [reply Backend/src/main/java/ktsnvt/tim1/repositories/EventRepository.java]: Convert it to JPQL query instead, use (argument is null or argument = property) for possible null arguments

**Pull request merged**